File: java_pro/src/main/java/jv/com/tencent/cain/design_pattern/decorator/TShirts.java
package jv.com.tencent.cain.design_pattern.decorator;

/**
 * @author cainjiang
 * @date 2019-03-31
 * 具体的装饰者
 */
public class TShirts extends Finery {
    @Override
    protected void show() {
        System.out.print("大T恤 ");
        super.show();
    }
}

File: java_pro/src/main/kotlin/kt/com/tencent/cain/desigin_pattern/decorator/TShirts.kt
package kt.com.tencent.cain.desigin_pattern.decorator

/**
 * @author cainjiang
 * @date 2019-03-31
 * 具体的装饰者
 */
class TShirts : Finery() {
    override fun show() {
        print("T恤 ")
        super.show()
    }
}